app/javascript/packages/components/text-input.tsx

```diff
@@ -20,19 +25,25 @@ export interface TextInputProps extends InputHTMLAttributes<HTMLInputElement> {
 }
 
 function TextInput(
-  { label, id, className, ...inputProps }: TextInputProps,
+  { label, hint, id, className, ...inputProps }: TextInputProps,
   ref: ForwardedRef<HTMLInputElement>,
 ) {
   const instanceId = useInstanceId();
   const inputId = id ?? `text-input-${instanceId}`;
+  const hintId = id ?? `text-input-hint-${instanceId}`;
   const classes = ['usa-input', className].filter(Boolean).join(' ');
 
   return (
     <>
       <label className="usa-label" htmlFor={inputId}>
         {label}
       </label>
-      <input ref={ref} className={classes} id={inputId} {...inputProps} />
+      {hint && (
+        <div id={hintId} className="usa-hint">
+          {hint}
+        </div>
+      )}
+      <input aria-describedby={hintId} ref={ref} className={classes} id={inputId} {...inputProps} />
```

Comment:
This will assign the `aria-describedby` even if the hint isn't rendered. I _think_ this is okay (in my own testing it's not shown to be an issue), but wanted to flag anyways, in case we want to be extra conservative and avoid assigning the attribute unless the hint is present.

Reply:
I think we can set the value to `undefined` to make sure it's left off if we don't want it:

```tsx
<input aria-describedby={hint ? hintId : undefined} />
```

If we're okay with some "clever"-ness for the sake of brevity, we could probably take advantage of how `&&` works with falsey left-hand values when `hint` itself is `undefined`.

```tsx
<input aria-describedby={hint && hintId} />
```